picker_wrapper.go

```diff
@@ -40,10 +42,18 @@ type pickerWrapper struct {
 	// The latest connection happened.
 	connErrMu sync.Mutex
 	connErr   error
+
+	stickinessMu    sync.Mutex
+	stickinessMDKey string
```

Comment:
If this can change after initialization, how about using `atomic.Value` instead of mutex?
Since `stickyStore` has a mutex inside of it, we don't need another mutex to protect it.